infrastructure/builder/src/generate/generators/k8s.js

```diff
@@ -0,0 +1,354 @@
+  ui: {
+    vars: [],
+    procs: {
+      web: {
+        type: 'web',
+        readinessPath: '/',
+        paths: [
+          '/*',
+        ],
+      },
+    },
+  },
+  references: {
+    vars: [],
+    procs: {
+      web: {
+        type: 'web',
+        readinessPath: '/references/',
+        paths: [
+          '/references',
+          '/references/*',
+          '/schemas',
+          '/schemas/*',
+        ],
+      },
+    },
+  },
+};
+Object.entries(extras).forEach(([name, {procs, vars}]) => {
+  exports.tasks.push({
+    title: `Generate helm templates for ${name}`,
+    requires: ['k8s-templates'],
+    provides: [`ingresses-${name}`],
+    run: async (requirements, utils) => {
+      const templates = requirements['k8s-templates'];
+      return {
+        [`ingresses-${name}`]: await renderTemplates(name, vars.map(v => v.var), procs, templates),
+      };
+    },
+  });
+});
+
+exports.tasks.push({
+  title: `Generate ingress`,
+  requires: ['k8s-templates', 'ingresses-ui', 'ingresses-references', ...SERVICES.map(name => `ingresses-${name}`)],
+  provides: [],
+  run: async (requirements, utils) => {
+    const ingresses = [];
+    for (const [name, req] of Object.entries(requirements)) {
+      if (name.startsWith('ingresses-')) {
+        for (const ingress of req) {
+          for (const path of ingress.paths) {
+            ingresses.push({
+              path,
+              projectName: ingress.projectName,
+            });
+          }
+        }
+      }
+    }
+    const templates = requirements['k8s-templates'];
+    const rendered = jsone(templates['ingress'], {ingresses});
+    await writeRepoYAML(path.join(TMPL_DIR, 'ingress.yaml'), rendered);
+  },
+});
+
+exports.tasks.push({
+  title: `Generate values.yaml and values.schema.yaml`,
+  requires: [...SERVICES.map(name => `configs-${name}`), ...SERVICES.map(name => `procslist-${name}`)],
+  provides: [],
+  run: async (requirements, utils) => {
+    const schema = {
+      '$schema': 'http://json-schema.org/draft-06/schema#',
+      type: 'object',
+      title: 'Taskcluster Configuration Values',
+      properties: {
+        rootUrl: {
+          type: 'string',
+          format: 'uri',
+          description: 'The url pointing to your cluster ingress.',
+        },
+        dockerImage: {
+          type: 'string',
+          description: 'The docker image containing taskcluster.',
+        },
+        ingressStaticIpName: {
+          type: 'string',
+          description: 'A google static ip object that the ingress can use to maintain the same ip address.',
+        },
+        ingressCertName: {
+          type: 'string',
+          description: 'A google certificate name that the ingress can use to set up tls.',
+        },
+      },
+      required: ['rootUrl', 'dockerImage'],
+      aditionalProperties: false,
+    };
+
+    // Something to copy-paste for users
+    const exampleConfig = {
+      rootUrl: '...',
+      dockerImage: '...',
```

Comment:
I liked that this sort of "automatically" selected the TC docker image version corresponding to the tf templates.  Is it possible to do the same here, to avoid the situation where users update the docker image and helm templates out-of-sync?  Maybe with some Helm/Go-Template conditionals to use the "official" docker image if an override isn't given?

Reply:
I've now made it so that `yarn release` updates the default value in `values.yaml` for the helm chart. It should fall back to the most recent release if not overridden.